dev/PushNotifications/PushNotificationUtility.h

```diff
@@ -127,9 +130,18 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
     }
     CATCH_RETURN()
 
+    inline void ClearRoActivateInstanceCache()
+    {
+        // Activating a nonexistent class clears the cache enables the proxyStub to be found for elevated scenarios
+        auto activatableClass{ wil::make_unique_string_nothrow<wil::unique_hstring>(L"FakeClass.ClassName") };
```

Comment:
What is the perf overhead of doing this? Is there a bug to fix the root cause here that we need to add to comments?

Reply:
OS bug is 39521331. Vestiges of caching logic predating dynamic dependencies cause WinRT proxy/stub creation paths to overlook the fact that the package graph has become non-empty.

As far as perf goes, the bug only requires you to do this fake activation _once_ after dynamic dependencies have taken the package graph size from 0 -> 1. So if this code is only ever run after dynamic dependencies have been added, then you could just make ClearRoActivateInstanceCache do the bogus activation once, and then no-op on all subsequent calls.

Reply:
Thanks. Then wouldn't dynamic dependencies be a better spot to add this rather than notifications? I imagine there may be other feature sets that are also impacted here and could use the goodness.

Reply:
> Then wouldn't dynamic dependencies be a better spot to add this rather than notifications?
That probably would be a little bit cleaner. @DrusTheAxe, thoughts?

Reply:
I'm not sure if this is correct, but I thought packaged apps don't need DynamicDependencies. And this fix is for both packaged/unpackaged apps so I don't think it can go in DynDep.

Reply:
What permutations of app+WinAppSDK need this?

![image](https://user-images.githubusercontent.com/72222/195904895-be30a659-c223-4194-848c-541b802d5039.png)

Reply:
Packaged+Unpackaged | MSIX. We don't support SelfContained since we can't guarantee the PushNotificationsLRP is installed.

Reply:
>the bug only requires you to do this fake activation once after dynamic dependencies have taken the package graph size from 0 -> 1

Is this only true of WinAppSDK's DynDep or also true for the [Win11 OS APIs](https://learn.microsoft.com/windows/win32/api/appmodel/nf-appmodel-addpackagedependency)?

Reply:
The reason packaged wasn't working in the first place because it couldn't get the proxyStub from a trusted source. This PR adds the proxyStub to the WASfwk which is the fix for packaged. I just mixed up the two issues. Oops.

This means only unpackaged | MSIX (WAS) is broken due to the bug.

Reply:
> Is this only true of WinAppSDK's DynDep or also true for the [Win11 OS APIs](https://learn.microsoft.com/windows/win32/api/appmodel/nf-appmodel-addpackagedependency)?

The bug is in the OS code, so it definitely affects the docked APIs. I'm pretty sure that WinAppSDK's DynDep doesn't manage to avoid the problem either, since IIRC the undocked DynDeps are focused primarily on activation, and don't do anything special for proxy/stub creation (which is reasonable; proxy/stub creation is way less straightforward to detour than inproc activation).

> If the bug is only when package graph goes from empty to not-empty (0->1) then why does Packaged need this?

Yeah, I don't follow that either. There are other parts of this change that are definitely relevant to both packaged and unpackaged apps, but the issue that ClearRoActivateInstanceCache is mitigating should only apply to processes that initially have an empty package graph.

Reply:
@pmpurifoy What about RS5 scenario for MSIX. Don't they also rely on the Long Running Process to activate the app in question?

Reply:
As long as the packaged app has a declared dependency on the WASfwk package that contains the proxyStub, it will work since now it can trust the source of the proxyStub.

Reply:
>As far as perf goes, the workaround for the bug only requires you to do this fake activation once 

Yes, but you didn't answer the question. MddBootstrapInitialize is likely the DynDep caller causing the pkggraph 0->1 transition. This is one of the 1st things an unpackaged app does in main and that's always a hotpath. Do you have a sense of the perf impact?

Doing this ClearRoCache in DynDep would be a tax on all unpackaged apps' critical startup path is one reason to do it in PushNotifications - defer it until needed so it's not in app startup hotpath, and only those who need it take the hit (Pay-to-Play)

If/when others besides PushNotifications need this we could move `ClearRoActivateInstanceCache()` to `dev\Common\Microsoft.RoApi.h` for reuse.

Reply:
> As far as perf goes, the workaround for the bug only requires you to do this fake activation once

Perf impact would primarily be loading the private hive (activationStore.dat). The hive remains loaded, so it's a one-time cost. That's on top of any cost payed by the undocked detoured RoActivateInstance before reaching the docked code. That's definitely a cost that will be paid eventually, but I can't say for certain whether that happens within the startup critical path right now.

In terms of total perf cost (i.e. ignoring the fact that the critical path is critical), it would arguably be best to do this right after the 0 -> 1 transition, rather than waiting to some later point where the package graph size may be > 1. That way we're loading/querying as few packages as possible before failing. Not sure that matters in practice, if nobody other than MddBootstrapInitialize is setting up DynDeps for these apps.

Pay-to-play is almost certainly better for perf. The risk is that it's creating a potential pitfall for any other WASDK components that try to use WinRT-style proxy registrations on systems without the OS bugfix. So that's basically a maintainability/development risk weighed against a perf risk. I don't think either risk is prohibitive.

Reply:
I just moved the function to dev/common/Microsoft.RoApi.h with a use case description for all WAS components to pay-to-play. Let me know if anything else is needed for this PR to be approved 😄